google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -218,7 +218,12 @@ public void apply() throws ConfigurationException {
     }
     sdkServiceUserSettings.setUserSelectedSdkServiceType(selectedCloudSdkServiceType);
 
+    boolean previousAutomaticUpdateEnabled = sdkServiceUserSettings.isAutomaticUpdateEnabled();
     sdkServiceUserSettings.setEnableAutomaticUpdates(enableAutomaticUpdatesCheckbox.isSelected());
+    if (enableAutomaticUpdatesCheckbox.isSelected() && !previousAutomaticUpdateEnabled) {
+      // activate updates again.
+      ManagedCloudSdkUpdateService.getInstance().activate();
```

Comment:
I'm trying to think of a scenario where this mechanism could be broken, potentially causing the update to be sceduled multiple times. For example, if the user has multiple projects opened and the user clicks to update on one of them, and then does the same on the other project (before it expires). Have you considered this scenarios to ensure everything happens atomically?

Reply:
Confirmed - notifications are synced in multiple project windows and only active one shows balloon. Once action is expired, it expires in all windows. `ManagedSDKService` itself also forbids simultaneous running of update/install jobs.